doc/classes/HMACContext.xml

```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="HMACContext" inherits="Reference" version="4.0">
+	<brief_description>
+	Used to create an HMAC for a message using a key.
+	</brief_description>
+	<description>
+	The HMACContext class is useful for advanced HMAC use cases, such as streaming the message as it supports creating the message over time rather than providing it all at once.
+	[codeblocks]
+	[gdscript]
+	extends Node
+	var ctx = HMACContext.new()
+
+	func _ready():
+		var key = "supersecret".to_utf8()
+		var err = ctx.start(HashingContext.HASH_SHA256, key)
+		assert(err == OK)
+		var msg1 = "this is ".to_utf8()
+		var msg2 = "vewy vewy secret".to_utf8()
+		err = ctx.update(msg1)
+		assert(err == OK)
+		err = ctx.update(msg2)
+		assert(err == OK)
+		var hmac = ctx.finish()
+		print(hmac.hex_encode())
+
+	[/gdscript]
+	[csharp]
```

Comment:
@Calinou @akien-mga I'm not super familiar with Godot's C# jazz. Mind checking my work here?